errai-marshalling/src/main/java/org/jboss/errai/marshalling/rebind/api/impl/defaultjava/DefaultJavaMappingStrategy.java

```diff
@@ -262,11 +269,25 @@ else if (instantiationMapping instanceof FactoryMapping) {
           builder._(loadVariable("a1").invoke("recordObject",
               loadVariable("objId"), loadVariable("entity")));
         }
+        
+        CaseBlockBuilder switchBlock;
+        builder.append(StatementBuilder.create()
+            .loadVariable("obj").invoke("keySet")
+            .foreach("key", String.class)
```

Comment:
@slavap The error stacktrace points to this line. For some reason Errai Codegen thinks the actual type of the loop variable is `Object`.

I would suggest working around this by chainging this line to:
    .foreach("key", Object.class)

Then use `Stmt.cast` below where necessary.